From 960cc06b5cc1ede58cabf98d5a16cd5011cb4ec6 Mon Sep 17 00:00:00 2001
From: Wang Sheng-Hui <shhuiw@gmail.com>
Date: Tue, 29 May 2012 15:06:17 -0700
Subject: [PATCH] mm/hugetlb.c: use long vars instead of int in region_count()

References: Hugetlb Fault Scalability (fate#317271)
Patch-mainline: v3.5
Git-commit: f2135a4a57a798ca8014a138afb626b96d90c842

The arguments f & t and fields from & to of struct file_region are
defined as long.  So use long instead of int to type the temp vars.

Signed-off-by: Wang Sheng-Hui <shhuiw@gmail.com>
Acked-by: David Rientjes <rientjes@google.com>
Acked-by: Hillf Danton <dhillf@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 mm/hugetlb.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/hugetlb.c b/mm/hugetlb.c
index 501e583..001d9fb 100644
--- a/mm/hugetlb.c
+++ b/mm/hugetlb.c
@@ -277,8 +277,8 @@ static long region_count(struct resv_map *resv, long f, long t)
 
 	/* Locate each segment we overlap with, and count that overlap. */
 	list_for_each_entry(rg, head, link) {
-		int seg_from;
-		int seg_to;
+		long seg_from;
+		long seg_to;
 
 		if (rg->to <= f)
 			continue;
